kaggle_environments/envs/hungry_geese/hungry_geese.py

```diff
@@ -264,21 +272,21 @@ def interpreter(state, env):
             for goose in geese
             for position in goose
         }
-        available_positions = {
-            i for i in range(rows * columns)
-            if (
-                i not in collisions and
-                i not in food
-            )
-        }
+        available_positions = set(range(rows * columns)).difference(collisions).difference(food)
+        # Ensure we don't sample more food than available positions.
+        needed_food = min(needed_food, len(available_positions))
         food.extend(sample(available_positions, needed_food))
 
+    # Set rewards after deleting all geese to ensure that geese don't receive a reward on the turn they perish.
```

Comment:
nit: is this comment valid, given the `agent.status == "ACTIVE"` check anyway?

Reply:
Yes, at the start of the turn geese may have a status of ACTIVE that gets changed to DONE before this step, e.g. due to collision. This comment is indicating that this block cannot be moved or else an agent might receive a score update on the same turn we set the status to DONE. This is problematic if two geese remain and one goose crashes into the body of the other. If they both receive score updates on the final turn they'll end in a tie even though one goose dies. If only the surviving agent gets the final score update, the surviving agent wins.

Reply:
Thanks, my misunderstanding.  Perhaps the comment could say that this needs to be between checking for collisions and checking for winners?  But pretty nitpicky